physics/smoke_dust/module_add_emiss_burn.F90

```diff
@@ -89,6 +89,8 @@ subroutine add_emis_burn(dtstep,dz8w,rho_phy,pi,                  &
     if (ebb_dcycle==2) then
 
     ! Constants for the fire diurnal cycle calculation
+     coef_con = 1._kind_phys/((2._kind_phys*pi)**0.5_kind_phys * sigma_fire_dur(1) *fire_age) * &
```

Comment:
This isn't equivalent to what it was before it was changed. I was referring to only moving `1._kind_phys/((2._kind_phys*pi)**0.5_kind_phys` outside of the loop since it doesn't need to be recalculated for every point. Division and exponentiation are relatively expensive computationally, so you should calculate this outside the loop since it doesn't change. The rest of the expression depends on `fire_age`, which is dependent on the location, and needs to stay in the loop.

Reply:
@grantfirl  It looks like 1._kind_phys/((2._kind_phys*pi)**0.5_kind_phys could not be separated from the remainings. The dividend is 1, and the divisor is the whole terms within the bracket as,
                ((2._kind_phys*pi)**0.5_kind_phys * sigma_fire_dur(1) *fire_age) * &
                exp(- ( log(fire_age) - avg_fire_dur(1))**2 /(2._kind_phys*sigma_fire_dur(1)**2 ))

Since sigma_fire_dur(1) and sigma_fire_dur(1) are loop independent, and we can move it outside of the loop. I think our current form is reasonable to save computing time. Thanks.

 57    real(kind_phys), dimension(1:5), parameter :: avg_fire_dur   = (/8.9, 4.2, 3.3, 3.0, 1.4/)
 58    real(kind_phys), dimension(1:5), parameter :: sigma_fire_dur = (/8.7, 6.0, 5.5, 5.2, 2.4/)

Reply:
@grantfirl    It looks like 1._kind_phys/((2._kind_phys*pi)**0.5_kind_phys could not be separated from the remainings. The dividend is 1, and the divisor is the terms within the bracket as
((2._kind_phys*pi)**0.5_kind_phys * sigma_fire_dur(1) *fire_age)

Probably we have to restore to its original position. Thanks.

Reply:
It's not that big of a deal since we're only talking about a minor potential speedup, but isn't 1/(A*B) equivalent to 1/A*1/B where A = sqrt(2*pi) and B = sigma_fire_dur*fire_age? That is, you could calculate 1/A outside of the loop and 1/B inside of the loop. It may not be worth it, and it can all just stay inside the loop if that makes more sense to do. It's up to you.

Reply:
@grantfirl Thanks for your explanation, and I understand your meaning. The code will be updated as,

     coef_con=1._kind_phys/((2._kind_phys*pi)**0.5_kind_phys)
     do j=jts,jte
       do i=its,ite
        fire_age= time_int + (fire_end_hr(i,j)-1._kind_phys)*3600._kind_phys  !One hour delay is due to the latency of the RAVE files
        fire_age= MAX(0._kind_phys,fire_age)

          SELECT CASE ( fire_type(i,j) )   !Ag, urban fires, bare land etc.
          CASE (1)
             ! these fires will have exponentially decreasing diurnal cycle,
             coef_bb_dc(i,j) = coef_con*1._kind_phys/(sigma_fire_dur(1) *fire_age) *                           &
                             exp(- ( log(fire_age) - avg_fire_dur(1))**2 /(2._kind_phys*sigma_fire_dur(1)**2 ))